src/main/fc/fc_msp.c

```diff
@@ -748,16 +748,30 @@ static bool mspCommonProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst, mspPostProce
         sbufWriteU8(dst, batteryConfig()->currentMeterSource);
         break;
 
-    case MSP_TRANSPONDER_CONFIG:
+    case MSP_TRANSPONDER_CONFIG: {
 #ifdef TRANSPONDER
-        sbufWriteU8(dst, 1); //Transponder supported
-        for (unsigned int i = 0; i < sizeof(transponderConfig()->data); i++) {
-            sbufWriteU8(dst, transponderConfig()->data[i]);
+        sbufWriteU8(dst, TRANSPONDER_PROVIDER_COUNT);
```

Comment:
Yes. But MSP is kept backwards compatible unless this is completely impossible. I.e. new data is always appended to the existing message format. The convention is that both sides only read the bytes of a message that are defined in the API version that they implement, and ignore trailing 'extra' bytes. This way, things (not necessarily the communicator) can talk to the firmware even if it is newer than the API version they are supporting. With respect to BOXNAMES, this will probably be relevant for the umpteen OSD implementations out there that try to read and display the active mode.